magicbot/state_machine.py

```diff
@@ -361,7 +361,10 @@ def next_state(self, name):
         
         .. note:: This should only be called from one of the state functions
         '''
-        state = self.__states[name]
+        if type(name) is str:
```

Comment:
`isinstance(name, str)` is preferred (allows subclasses).

However, I think I would prefer an explicit check for the function, rather than a check for the name. Otherwise, you could accidentally pass a different type of object in and the user would just get strange errors.

I think a preferred embodiment of this would add all of the state methods to a set at the beginning, and then this code is just something like:

```
if name_or_state in self.__statemethods:
    state = name_or_state
else:
    state = self.__states[name]
```